core/src/test/kotlin/fr/sncf/osrd/stdcm/preprocessing/BlockAvailabilityTests.kt

```diff
@@ -268,12 +272,35 @@ class BlockAvailabilityTests {
             availability.getAvailability(
                 explorer,
                 Offset(0.meters),
-                explorer.getSimulatedLength(),
+                explorer.getSimulationEndPathOffset(),
                 0.0
             ) as BlockAvailabilityInterface.Available
         assertEquals(minStartTime, res.timeOfNextConflict)
     }
 
+    /**
+     * In this case Offset<Path> != Offset<EnvelopePath>, checks that offsets types aren't mixed up
```

Comment:
As there is ongoing work around naming, `EnvelopePath` doesn't exist so I think we shouldn't introduce something new.
It looks like current type for envelope sequence is `EnvelopeConcat`.
`EnvelopeSimPath` exists, so it may be used, even though it's not a sequence of designated object.
We may come up with `EnvelopePartPath` to have a sequence of designated object?

Not sure about the meaning here, but maybe just `Offset<Envelope>` ?

Reply:
In the current implementation, in every case `Offset<Envelope> == Offset<TravelledPath>`.

But it's true that it's not *necessarily* the case, and we could introduce that type.

There's a slight ambiguity though: for an envelope ranging from position 50 to 100, where does `Offset<Envelope>(50.meters)` point to? Is it the start or the end?

Reply:
In my understanding of `Object`Path is a sequence of instances of `Object`: it points to the end, because it's not a path.

Anyway, I think this comment should move a bit, as I still don't understand if it refers to EnvelopeConcat or Envelope:
```suggestion
     * In this case Offset<Path> != Offset<EnvelopeConcat>, checks that offsets types aren't mixed up
```
?

Reply:
Oh I missunderstood the initial comment (specifically I didn't read my own code carefully enough).

I intended to write `Offset<TravelledPath>` there, and I agree we can stick to well established types even in comments. Even if envelope offset typing would be nice.

Reply:
I've changed it to `TravelledPath`, is it OK for you? I feel like I'm dodging the rest of the discussion, but I don't know if it's still relevant.

Reply:
I'm perfectly OK to leave the rest of the discussion to https://github.com/OpenRailAssociation/osrd/issues/10239 (and related work).
Thanks!